src/internal/operators/elementAt.ts

```diff
@@ -1,11 +1,14 @@
 import { ArgumentOutOfRangeError } from '../util/ArgumentOutOfRangeError';
 import { Observable } from '../Observable';
-import { MonoTypeOperatorFunction } from '../types';
+import { MonoTypeOperatorFunction, OperatorFunction } from '../types';
 import { filter } from './filter';
 import { throwIfEmpty } from './throwIfEmpty';
 import { defaultIfEmpty } from './defaultIfEmpty';
 import { take } from './take';
 
+export function elementAt<T>(index: number): MonoTypeOperatorFunction<T>;
+export function elementAt<T, D>(index: number, defaultValue: D): OperatorFunction<T, T | D>;
+
```

Comment:
nitpick: Why are these overloads necessary? They don't seem to do anything other than make `defaultValue` option - which it already is in the implementation signature.

Reply:
I can see removing this breaking folks specifying one generic and not using defaultValue. But we don't care about that use case anymore IIRC

Reply:
It's required. Otherwise this fails, and you get `Observable<unknown>` because `D` is `unknown`, I guess.

```ts
  const o = of('foo').pipe(elementAt(47)); // $ExpectType Observable<string>
```

Reply:
https://github.com/ReactiveX/rxjs/blob/c6ba9790ed42e4781278822aff96bce0863e7917/src/internal/operators/last.ts#L12